ecoscope/io/earthranger.py

```diff
@@ -922,3 +798,94 @@ def patch_event(
         else:
             results = [self._patch(f"activity/event/{event_id}", payload=events)]
         return pd.DataFrame(results)
+
+    def post_subject(
+        self,
+        subject_name: str,
+        subject_type: str,
+        subject_subtype: str,
+        is_active: bool = True,
+        tracks_available: bool = True,
+        **kwargs,
+    ) -> pd.DataFrame:
+        """
+        Parameters
+        ----------
+        subject_name
+        subject_type
+        subject_subtype
+        is_active
+        tracks_available
+
+        Returns
+        -------
+        pd.DataFrame
+        """
+
+        payload = {
+            "name": subject_name,
+            "subject_type": subject_type,
+            "subject_subtype": subject_subtype,
+            "is_active": is_active,
+            "tracks_available": tracks_available,
+        }
+
+        if kwargs:
+            payload.update(kwargs)
+
+        response = self._post("subjects", payload=payload)
+        return pd.DataFrame([response])
+
+    def post_source(
+        self,
+        source_type: str,
+        manufacturer_id: str,
+        model_name: str,
+        provider: str,
```

Comment:
does this require the provider ID value? Can it be null?

Reply:
No : only the name. Provider is a required parameter to create a source from the API, but I modified the function so that "default" is the provider name if the user does not specify it.